fuzzamoto-ir/src/compiler.rs

```diff
@@ -57,18 +57,27 @@ pub struct CompiledProgram {
     pub metadata: CompiledMetadata,
 }
 
+pub type VariableIndex = usize;
+
+pub type InstructionIndex = usize;
+
+pub type ActionIndex = usize;
+
 #[derive(serde::Serialize, serde::Deserialize, Debug, Clone)]
 pub struct CompiledMetadata {
     // Map from blockhash to (block variable index, list of transaction variable indices)
     block_tx_var_map: HashMap<bitcoin::BlockHash, (usize, Vec<usize>)>,
+    // Map mapping from action indices to the connection var id.
+    send_action_map: HashMap<ActionIndex, VariableIndex>,
```

Comment:
I was more thinking of having something like a map from connection id to connection variable index. So for every 

`vN <- LoadConnection(x)`

would create an entry in the map, mapping `x` to `N`.

And in the scenario when we receive messages we know the connection id and can look up the variable index.